net/respondd-module-lldp/src/respondd.c

```diff
@@ -0,0 +1,51 @@
+#include <respondd.h>
+#include <json-c/json.h>
+#include <lldpctl.h>
+#include <lldp-const.h>
+
+static struct json_object * respondd_provider_neighbours(void) {
+    lldpctl_conn_t *conn;
+    lldpctl_atom_t *ifaces, *iface, *port, *neighbors, *neighbor;
+    const char *ctlname, *neighmac, *portmac;
+    struct json_object *ret, *ret_lldp, *neighbors_obj;
+
+    ret_lldp = json_object_new_object();
+
+    ctlname = lldpctl_get_default_transport();
+    conn    = lldpctl_new_name(ctlname, NULL, NULL, NULL);
+    ifaces  = lldpctl_get_interfaces(conn);
+    lldpctl_atom_foreach(ifaces, iface) {
+        port = lldpctl_get_port(iface);
+        // check if Port ID Subtype is MAC address
+        if (lldpctl_atom_get_int(port, lldpctl_k_port_id_subtype) != LLDP_PORTID_SUBTYPE_LLADDR)
+            continue;
+
+        portmac = lldpctl_atom_get_str(port, lldpctl_k_port_id);
+        if (!portmac)
+            continue;
+
+        neighbors_obj = json_object_new_object();
+        neighbors = lldpctl_atom_get(port, lldpctl_k_port_neighbors);
+        lldpctl_atom_foreach(neighbors, neighbor) {
+            // check if Chassis ID Subtype is MAC address
+            if (lldpctl_atom_get_int(neighbor, lldpctl_k_chassis_id_subtype) != LLDP_CHASSISID_SUBTYPE_LLADDR)
+                continue;
+
+            neighmac = lldpctl_atom_get_str(neighbor, lldpctl_k_chassis_id);
+            if (!neighmac)
+                continue;
+
+            json_object_object_add(neighbors_obj, neighmac, json_object_new_object());
```

Comment:
It seems you are adding an empty object here? So the "name" and "descr" fields described in the README don't actually exist?

Reply:
done